src/Extractor/DibiSupportExtractorTrait.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Extractor;
+
+use Dibi;
+use Dibi\Connection;
+use Keboola\Csv\CsvFile;
+use Keboola\Csv\Exception as CsvException;
+use Keboola\DbExtractor\Exception\ApplicationException;
+use Keboola\DbExtractor\Exception\DeadConnectionException;
+use Keboola\DbExtractorLogger\Logger;
+use Throwable;
+use Dibi\Result;
+use Keboola\DbExtractor\DbRetryProxy;
+use Keboola\DbExtractor\Exception\UserException;
+
+/**
+ * This trait solves compatibility problem:
+ * - Extractor class from db-extractor-common depends on PDO
+ * - PDO ODBC driver (DIBI -> PDO -> ODBC -> DB) doesn't work with special types (MAP, ARRAY, ...) -> it returns NULL
+ * - ODBC driver (DIBI -> ODBC -> DB) works with special types
+ * - ... therefore must be used ODBC driver and fixed compatibility with db-extractor-common package
+ */
+trait DibiSupportExtractorTrait
```

Comment:
Trocha sa to cele skomplikovalo.

Napisal som testy na specialne typy (`map`, `array`, `binary`, ...) z Hive Db, a zistil som, ze aj ked je to po stranke kodu napisane spravne, tak to nejde. Skusal som aj ine verzie ODBC Driverov, .... a nakoniec som zistil, ze problem je v `pdo_odbc` extension. 

- S `DIBI -> **PDO** -> ODBC -> Hive DB` specialne typy nejdu (low-level funkcia vrati `NULL`), ... bud je tam bug, alebo to nie je implementované
- ... ale s `DIBI -> ODBC -> Hive DB` to ide, takze som to musel cele prerobit.

`db-extractor-common` je zalozeny na PDO, takze vznikol tento skaredy trait `DibiSupportExtractorTrait`.

Ine riesenie asi nie je .... okrem toho, ze by sa `db-extractor-common` cely prepisal, co sa asi aj tak stane (je tam nejaky nedokonceny PR), ale v tomto sprinte urcite nie, ... takze som tam vymyslel workaround.

Reply:
Cela zmena: https://github.com/keboola/db-extractor-hive/pull/2/commits/af12f436523763e3b235e623eaf5591827ece5f1